ERCS/erc-7579.md

```diff
@@ -48,149 +48,164 @@ The smart account's `validateUserOp` function SHOULD return the return value of
+     * @dev Executes a transaction on behalf of the account.
+     *         This function is intended to be called by Executor Modules
+     * @param mode The encoded execution mode of the transaction. See ModeLib.sol for details
+     * @param executionCalldata The return data of the executed contract call
+     *
+     * MUST implement this function
+     * MUST ensure adequate authorization control: i.e. onlyExecutorModule
+     * If a mode is requested that is not supported by the Account, it MUST revert
+     */
+    function executeFromExecutor(bytes32 mode, bytes calldata executionCalldata) external payable;
 
     /**
-    * MUST execute a `call` to all the targets with the provided data and value for each target
-    * MUST allow ERC-4337 Entrypoint to be the sender and MAY allow `msg.sender == address(this)`
-    * MUST revert if any call was not successful
-    */
-    function executeBatch(Execution[] calldata executions) external returns (bytes memory result);
+     * @dev ERC-4337 executeUserOp according to ERC-4337 v0.7
+     *         This function is intended to be called by ERC-4337 EntryPoint.sol
+     * @param userOp PackedUserOperation struct (see ERC-4337 v0.7+)
+     *
+     * MAY implement this function
+     * MUST ensure adequate authorization control: i.e. onlyEntryPointOrSelf
+     */
+    function executeUserOp(UserOperation calldata userOp) external payable;
+}
+```
+
+The execution mode is a `bytes32` value that is structured as follows:
+
+- callType (1 byte): `0x00` for a single `call`, `0x01` for a batch `call` and `0xff` for `delegatecall`
+- execType (1 byte): `0x00` for a normal execution, `0x01` for a try/catch execution
+- unused (4 bytes): this range is reserved for future standardization
+- modeSelector (4 bytes): an additional mode selector that can be used to create further execution modes
+- modePayload (22 bytes): additional data to be passed
+
+Here is a visual representation of the execution mode:
+
+| CallType | ExecType | Unused  | ModeSelector | ModePayload |
+| -------- | -------- | ------- | ------------ | ----------- |
+| 1 byte   | 1 byte   | 4 bytes | 4 bytes      | 22 bytes    |
+
+#### Account configurations
+
+To comply with this standard, smart accounts MUST implement the account config interface below:
 
+```solidity
+interface IAccountConfig {
     /**
-    * MUST execute a `call` to the target with the provided data and value
-    * MUST only allow installed executors to call this function
-    * MUST revert if the call was not successful
-    */
-    function executeFromExecutor(address target, uint256 value, bytes callData) external returns (bytes memory result);
+     * @dev Initializes the account. Function might be called directly or by a factory
+     * @param data encoded data that can be used during the initialization phase
+     *
+     * MUST revert if the account is already initialized
+     */
+    function initializeAccount(bytes calldata data) external payable;
```

Comment:
Yeah this is a good point - it could make sense to remove this (wdyt @zeroknots). 

One use-case I see is initializing the same account but at different levels of modularity. Eg [this function](https://github.com/erc7579/erc7579-implementation/blob/d0cc778a5ad31efe43b507cf3b79e012c461a7de/src/utils/Bootstrap.sol#L16) could be called to create an account with a single validator which would be much cheaper gas-wise vs [this function](https://github.com/erc7579/erc7579-implementation/blob/d0cc778a5ad31efe43b507cf3b79e012c461a7de/src/utils/Bootstrap.sol#L21) could be used to init the account with multiple module types. Having different factories for different permutations of module types seems like an annoying operational problem

Reply:
We've removed this from the standard. I think it could still be useful for accounts (eg for reason mentioned above), but I agree that accounts should be able to decide how to implement this and standardization is not really useful here